Route-editing + waypoint view fields

ginold opened the pull request:
small update for route-editing form and wp view.

answers #277  and #280

asaunier commented:
Please use more explicit commit messages like "fix attributes incorrectly shown in route editing form" or "Move access_period to the route text attributes block".
Else we won't have much info about the changes when looking at the commits history.

I have just tested the route editing form: the extra MTB fields no longer show up when selecting skitouring. But neither when selecting MTB?

Thanks for the change about access_period!

asaunier commented:
Please merge if 
>  the extra MTB fields no longer show up when selecting skitouring. But neither when selecting MTB?

is addressed.

ginold commented:
i've made a few other small changes -> i used the mako syntax in for loops instead of angular, 
changed url to a <a href> link, deleted a field that i created but didn't even exist, added other one missing.. maybe you want to have a quick look.

But i'm still waiting to clarify the 'orientation(s)' question. I also still see some strange errors while saving some attributes, but i don't know if I should correct it or is it an API problem?

For example : 
`['multi', 'bloc'] is not a string: {'climbing_outdoor_types': ''}` or `"None" is not iterable` even if it can be null and is not required.

asaunier commented:
>  i'm still waiting to clarify the 'orientation(s)' question

See https://github.com/c2corg/v6_api/pull/227 (and in common)

> "None" is not iterable even if it can be null and is not required.

This seems to be a problem with the API, see https://github.com/c2corg/v6_ui/issues/110#issuecomment-208739547

> ['multi', 'bloc'] is not a string: {'climbing_outdoor_types': ''}

Are you sure that the error?
I guess the API expects to receive:
```
'climbing_outdoor_types': ['multi', 'bloc']
```

asaunier commented on c2corg_ui/templates/route/edit.html:
```diff
@@ -80,7 +80,7 @@ <h2 class="heading show-phone" translate>general informations</h2>
               </div>
             </div>
-            
+            {{route.orientations}}
```
Is this really needed?

asaunier commented on c2corg_ui/templates/route/helpers/detailed_route_attributes.html:
```diff
@@ -2,22 +2,20 @@
         <p translate>Location</p>
       </div>
-      <span class="detail-text accordion">
+      <span class="detail-text accordion" id="detail-orientation" ng-init="route.orientation = ${route.get('orientations')}">
```
Missing "s" in ``ng-init="route.orientation``?

asaunier commented on c2corg_ui/templates/route/helpers/detailed_route_attributes.html:
```diff
@@ -223,21 +240,23 @@
 ## ACCESS
 <%def name="get_route_access(route)">\
-  % if route.get('lift_access') or route.get('height_diff_access'):
+  % if route.get('lift_access') == False or route.get('lift_access') == True or route.get('height_diff_access'):
```
```
% if route.get('lift_access') not None or route.get('height_diff_access'):
```

asaunier commented on c2corg_ui/templates/route/helpers/detailed_route_attributes.html:
```diff
@@ -223,21 +240,23 @@
-          <b translate class="value-title">lift_access</b>
-          <input type="checkbox" checked="checked"><span></span>
+
+        % if route.get('lift_access') == True:
```
```
% if route.get('lift_access'):
```

asaunier commented on c2corg_ui/templates/route/helpers/detailed_route_attributes.html:
```diff
@@ -223,21 +240,23 @@
+
+        % if route.get('lift_access') == True:
+        <span x-translate class="value-title">lift_access</span> : <span translate class="value">yes</span><br>
+        % elif route.get('lift_access') == False :
```
```
% elif route.get('lift_access') not None:
```

asaunier commented on c2corg_ui/templates/waypoint/helpers/detailed_waypoint_attributes.html:
```diff
@@ -300,29 +316,34 @@
 ## EQUIPMENT  
 <%def name="get_waypoint_equipment(waypoint)">\
-  % if waypoint.get('matress_unstaffed') or  waypoint.get('gas_unstaffed') or waypoint.get('heating_unstaffed') or waypoint.get('blanket_unstaffed') :
+  % if waypoint.get('matress_unstaffed') == True or waypoint.get('matress_unstaffed') == False or  \
```
```
% if waypoint.get('matress_unstaffed') not None or  \
```
ditto for the following conditions.

ginold commented:
> I guess the API expects to receive:

`'climbing_outdoor_types': ['multi', 'bloc']` <

that's exactly what i'm sending. I understand it as if API was awaiting a string rather than an array, or maybe JSON.stringified array but that makes little sense.

asaunier commented:
Indeed, I have the same behaviour.

> I understand it as if API was awaiting a string rather than an array

Actually the API indeed expects a string:
https://github.com/c2corg/v6_api/blob/master/c2corg_api/models/route.py#L148
not a list of strings. Ditto in the google sheet,

But that consistent: a route cannot be of several types at the same time!

tsauerwein commented:
If `climbing_outdoor_types` is not an array it should be named `climbing_outdoor_type`.

ginold commented:
that's exactly what is coming up next in one of Alex'es pr!

asaunier commented on requirements.txt:
```diff
@@ -14,6 +14,6 @@ Markdown==2.6.5
 # for development use a local checkout
 # -e ../v6_common
-git+https://github.com/c2corg/v6_common.git@343e1ef
+git+https://github.com/c2corg/v6_common.git@b8e402d
```
The hash for c2corg_common with the fixes you suggested about some route/WP attributes is f825f8f

asaunier commented:
Except for the hash of c2corg_common, your changes look good to me too.

ginold commented:
I changed the hash. May i merge after travis has passed?

asaunier commented:
:+1:
